lib/Controller/ThreadController.php

```diff
@@ -111,4 +121,37 @@ public function delete(int $id): JSONResponse {
 
 		return new JSONResponse();
 	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @param string $id
+	 * @param int $accountId
+	 *
+	 * @return JSONResponse
+	 */
+	public function summarize(string $id, int $accountId): JSONResponse {
+		try {
+			$account = $this->accountService->find($this->currentUserId, $accountId);
+			$messages = $this->mailManager->getThread($account, $id);
+		} catch (ClientException $e) {
+			return new JSONResponse([], Http::STATUS_FORBIDDEN);
+		}
+
+		try {
+			$summary = $this->aiIntergrationsService->summarizeThread(
+				$id,
+				$messages,
+				$this->currentUserId,
+			);
```

Comment:
Calling this in a route is risky as it will likely exceed the PHP max execution time. Also, you might want to cache the output. :)